configs/config.go

```diff
@@ -77,6 +77,15 @@ type Config struct {
 	Version *version.Version
 }
 
+// ModuleRequirements represents the provider requirements for an individual
+// module, along with references to any child modules. This is used to
+// determine which modules require which providers.
+type ModuleRequirements struct {
+	Module       *Module
+	Requirements getproviders.Requirements
+	Children     map[string]*ModuleRequirements
+}
```

Comment:
This type and the method which returns an instance of it below are intended to be used in an upcoming improvement to the `init` command's diagnostics when providers fail to install. That's why the `Module` member is here, and also why it's in the `configs` package instead of `command`.

Reply:
thank you for this extra context!